packages/trpc/server/routers/viewer/teams/lazyLoadMembers.handler.ts

```diff
@@ -0,0 +1,194 @@
+
+type LazyLoadMembersHandlerOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+    prisma: PrismaClient;
+  };
+  input: TLazyLoadMembersInputSchema;
+};
+
+// This should improve performance saving already app data found.
+const appDataMap = new Map();
+
+const userSelect = Prisma.validator<Prisma.UserSelect>()({
+  username: true,
+  email: true,
+  name: true,
+  avatarUrl: true,
+  id: true,
+  bio: true,
+  disableImpersonation: true,
+  teams: {
+    select: {
+      team: {
+        select: {
+          slug: true,
+          id: true,
+        },
+      },
+    },
+  },
+  credentials: {
+    select: {
+      app: {
+        select: {
+          slug: true,
+          categories: true,
+        },
+      },
+      destinationCalendars: {
+        select: {
+          externalId: true,
+        },
+      },
+    },
+  },
+});
+
+export const lazyLoadMembersHandler = async ({ ctx, input }: LazyLoadMembersHandlerOptions) => {
+  const { prisma } = ctx;
+  const { cursor, limit, teamId, searchTerm } = input;
+
+  const canAccessMembers = await checkCanAccessMembers(ctx, teamId);
+
+  if (!canAccessMembers) {
+    throw new TRPCError({
+      code: "UNAUTHORIZED",
+      message: "You are not authorized to access this team",
+    });
+  }
+
+  const teamMembers = await prisma.membership.findMany({
+    where: {
+      teamId,
+      ...(searchTerm && {
+        user: {
+          OR: [
+            {
+              email: {
+                contains: searchTerm,
+              },
+            },
+            {
+              username: {
+                contains: searchTerm,
+              },
+            },
+          ],
+        },
+      }),
+    },
+    select: {
+      id: true,
+      role: true,
+      accepted: true,
+      user: {
+        select: userSelect,
+      },
+    },
+    cursor: cursor ? { id: cursor } : undefined,
+    take: limit + 1, // We take +1 as itll be used for the next cursor
+    orderBy: {
+      id: "asc",
+    },
+  });
+
+  let nextCursor: typeof cursor | undefined = undefined;
+  if (teamMembers && teamMembers.length > limit) {
+    const nextItem = teamMembers.pop();
+    nextCursor = nextItem?.id;
+  }
+
+  const members = await Promise.all(
+    teamMembers.map(async (member) => ({
+      ...member,
+      user: await UserRepository.enrichUserWithItsProfile({
+        user: member.user,
+      }),
+    }))
+  );
+
+  const membersWithApps = members.map((member) => {
+    const { credentials, profile, ...restUser } = member.user;
+    return {
+      ...restUser,
+      username: profile?.username ?? restUser.username,
+      role: member.role,
+      profile: profile,
+      organizationId: profile?.organizationId ?? null,
+      organization: profile?.organization,
+      accepted: member.accepted,
+      disableImpersonation: member.user.disableImpersonation,
+      bookerUrl: getBookerBaseUrlSync(profile?.organization?.slug || ""),
+      connectedApps: credentials?.map((cred) => {
```

Comment:
This is a pretty expensive procedure to go through. We don't need this for the org member's page and that would have the most amount of users. Can we save this when listing team members?

Reply:
We do need connectedApps in `packages/features/ee/teams/components/MemberListItem.tsx` to display the apps logo and name next to member's name.

Reply:
> Can we save this when listing team members?

Could you elaborate more? How should i save this info

Reply:
@joeauyeung We need to show connectedApps right  next to the member's name.

Reply:
Maybe we can load it in a separate call? Members list is critical, connected apps can have skeleton and load in later.

Reply:
Agreed with @hariombalhara. Lazy load non-critical info in a separate endpoint and initiate the call from the client. We're ok with showing a loading skeleton in client for non critical data.